pkg/services/rendering/rendering.go

```diff
@@ -27,12 +29,31 @@ type RenderingService struct {
 	grpcPlugin   pluginModel.RendererPlugin
 	pluginInfo   *plugins.RendererPlugin
 	renderAction renderFunc
+	domain       string
 
 	Cfg *setting.Cfg `inject:""`
 }
 
 func (rs *RenderingService) Init() error {
 	rs.log = log.New("rendering")
+
+	// ensure ImagesDir exists
+	err := os.MkdirAll(rs.Cfg.ImagesDir, 0700)
+	if err != nil {
+		return err
+	}
+
+	// set value used for domain attribute of renderKey cookie
+	if rs.Cfg.RendererUrl != "" {
+		// AppUrl has already been passed, it wont generate an error.
+		u, _ := url.Parse(setting.AppUrl)
+		rs.domain = u.Hostname()
```

Comment:
Think this will stop Grafana systems behind a reverse proxy to work. as the AppUrl is the public (reverse proxy url) and grafana does not have the credentials to go through that (for alerting and other scenarios) so need to access directly ad bypass auth proxy.

Reply:
That is true.  The only way to accommodate all use cases then is to add a separate `[rendering]` setting for the URL the renderer should load, ie `callback_url`.  If the `callback_url` is set to `""` (the default) we just use AppUrl.